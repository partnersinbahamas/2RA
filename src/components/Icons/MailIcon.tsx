import React from 'react';
import PropTypes from 'prop-types';
import Svg from './Svg';

type TProps = {
  className?: string;
  color?: string;
  style?: { [cssAttribute: string]: string };
  withBox?: boolean;
};

const MailIcon: React.FC<TProps> = ({ className, color, style, withBox }) => (
  <Svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox={withBox ? '-2 -3 22 18' : '0 0 18 12'}
    style={style}
    fill={color}
    className={className}
  >
    <path d="M0 3.421v8.578h18V3.421l-9 5.25z" />
    <path d="M0 0v1.5l9 5.25 9-5.25V0z" />
  </Svg>
);

MailIcon.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  style: PropTypes.any,
  withBox: PropTypes.bool,
};

MailIcon.defaultProps = {
  className: '',
  color: 'currentcolor',
  style: {},
  withBox: false,
};

export default MailIcon;
