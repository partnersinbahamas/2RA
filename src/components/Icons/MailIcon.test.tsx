import React from 'react';
import { render } from '@testing-library/react';
import MailIcon from './MailIcon';

test('MailIcon', () => {
  const { container } = render(<MailIcon />);
  expect(container).toMatchSnapshot();
});
