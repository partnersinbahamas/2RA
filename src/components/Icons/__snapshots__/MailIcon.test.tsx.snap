// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailIcon 1`] = `
<div>
  <svg
    class="sc-beySPh jroDkm"
    fill="currentcolor"
    viewBox="0 0 18 12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 3.421v8.578h18V3.421l-9 5.25z"
    />
    <path
      d="M0 0v1.5l9 5.25 9-5.25V0z"
    />
  </svg>
</div>
`;
